---
title: "Alzheimers Disease Brain Blood Samples"
author: "Janis Corona"
date: "1/17/2020"
output:
  html_document:
    df_print: paged
  pdf_document: default
  word_document: default
---

This analysis is done using the NCBI gene expression samples taken from a study that used the middle temporal gyrus side of the brain of frozen samples taken from Alzheimer's patients that totalled 78 samples of control and repeat microarray gene expression results. The file was able to be downloaded and unzipped complete with the platform gene symbol ID already attached. A separate file of the meta information that includes the age, gender, tissue type and disease as Alzheimer's Disease (AD) or control was made from the Series information for each sample. The age range for these samples of healthy and AD patients is from 70-95 years of age.

This study can be linked to with all the sample and meta information at: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE109887
The link to the 41 mb text file of series of samples with ID that is too large for github, or my github account type is: ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE109nnn/GSE109887/matrix/GSE109887_series_matrix.txt.gz 

Open the original Sample ID values text file and write out as a csv file.
```{r}
alz <- read.delim('GSE109887_series_matrix.txt', sep='\t', header=TRUE, na.strings=c('',' '), comment.char='!')

write.csv(alz,'alzheimerSamples.csv', row.names=FALSE)

```

```{r}
head(alz)
```


Lets read in the meta information for age, gender, and disease type as AD or control.
```{r}
meta <- read.csv('AlzheimerAgeGenderTissueSamplesMeta.csv', sep=',',
                 header=TRUE, na.strings=c('',' '))
```

```{r}
head(meta)
```


Transpose the meta to fields instead of rows
```{r}

meta1 <- as.data.frame(t(meta))
names <- as.character(meta$sampleID)
colnames(meta1) <- names
meta2 <- meta1[-1,]
head(meta2)
```


Separate the data into gender and control or AD data sets respectively in each gender and for both genders.

Data tables of males and females for meta only:
```{r}
row.names(alz) <- alz$ID_REF
alz1 <- alz[,-1]

row.names(meta) <- meta$sampleID
Meta <- meta[,-1]

names <- row.names(Meta)
Meta1 <- as.data.frame(t(Meta))
colnames(Meta1) <- names

fem <- grep('F', Meta1$gender)
mal <- grep('M', Meta1$gender)

Fem <- Meta1[fem,]
Mal <- Meta1[mal,]

Fem$sampleID <- as.factor(row.names(Fem))
Mal$sampleID <- as.factor(row.names(Mal))
```

Display the fields of these gender specific tables by meta information:
```{r}
colnames(Fem)
colnames(Mal)
```

The same indices of each row name is the same as the alz1, alzheimer table colnames indices. So, use those same female and male indices values from the regex commands to separate the alzheimer table into male and female tables.
```{r}
Females <- alz1[,fem]
Males <- alz1[,mal]
```



Data tables of control or AD for males and for females for meta only:
```{r}
femControl <- grep('control', Fem$disease)
malControl <- grep('control', Mal$disease)

femAD <- grep('AD', Fem$disease)
malAD <- grep('AD', Mal$disease)

FemCtrl <- Fem[femControl,]
MalCtrl <- Mal[malControl,]

FemAD <- Fem[femAD,]
MalAD <- Mal[malAD,]
```

Display the AD or Control tables by gender for meta information:
```{r}
colnames(FemCtrl)
colnames(FemAD)
colnames(FemAD)
colnames(MalAD)
```

Now use that same index information within each gender table derived from the alzheimer gene expression data:
```{r}
Females_AD <- Females[,femAD]
Females_control <- Females[,femControl]

males_AD <- Males[,malAD]
males_control <- Males[, malControl]
```

Write these last tables to csv file:
```{r}
write.csv(Females_AD, 'females_Alzheimers.csv', row.names=TRUE)
write.csv(Females_control, 'females_healthy_control.csv', row.names=TRUE)
write.csv(males_AD, 'males_Alzheimers.csv', row.names=TRUE)
write.csv(males_control, 'males_healthy_control.csv', row.names=TRUE)
```

What is the age range on these samples of healthy controls and AD patients?
```{r}
age1 <- range(as.numeric(as.character(FemAD$age)))
age2 <- range(as.numeric(as.character(FemCtrl$age)))
age3 <- range(as.numeric(as.character(MalAD$age)))
age4 <- range(as.numeric(as.character(MalCtrl$age)))

char <- c('The minimum age: ', 'The maximum age: ')

AD_fem <- paste0(char,age1, sep='')
AD_mal <- paste0(char,age3, sep='')
ctrl_fem <- paste0(char,age2, sep='')
ctrl_mal <- paste0(char, age4, sep='')

adf <- c('Alzheimer females: ')
adm <- c('Alzheimer males: ')
cf <- c('healthy females: ')
cm <- c('healthy males: ')

ranges <- c(adf,AD_fem, adm, AD_mal, cf, ctrl_fem, cm, ctrl_mal)
ranges
```

The age range in all of this data for these healthy and AD patients is age 70 to age 95 years of age.Just useful to know when cross comparing blood microarray samples across the females from our overweight females using epigallocatechin (EGCG) in their 40s, the males from Russia who are also in their 40s with or without myocardial infarction (MI), the hemochromatosis (iron toxicity) males and females averaging their 40s in years of age with or without iron toxicity, and the flu vaccinated blood samples of males and females also in their 40s with or without antibiotic treatment combined with flu vaccination. 


Lets do some data analysis of the means and fold change between the groups to create some additional statistical information on these Alzheimers blood samples and healthy controls of elderly patients. We will use dplyr for this and plot with ggplot2 later.

```{r, error=FALSE, message=F, warning=F}
library(dplyr)
```

First lets look at the control and the AD tables within the gender specific tables: Females_AD, Females_control, males_AD, males_control. Get the means of each gene and compare into one table. This and these tables of data were already cleaned to only have one gene per row, due to not having to merge the platform fields to the series table and duplicate genes for missing data in the additional rows like previous studies mentioned earlier required.

Get the row means of each table and add to each table.
```{r}
names <- row.names(Females_AD)
Females_AD1 <- rowMeans(Females_AD)
#colnames(Females_AD1) <- 'fem_AD_Mean'

Females_control1 <- rowMeans(Females_control)
#colnames(Females_control1) <- 'fem_ctrl_Mean'

males_AD1 <- rowMeans(males_AD)
#colnames(males_AD1) <- 'mal_AD_Mean'

males_control1 <- rowMeans(males_control)
#colnames(males_control1) <- 'mal_ctrl_Mean'
```

```{r}
means <- cbind(Females_AD1, Females_control1, males_AD1, males_control1)
Means <- as.data.frame(means)
colnames(Means) <- paste(colnames(Means),'_Mean',sep='')
str(Means)
```

Now for the fold change values between the Means of the female control and AD patients, then for the male control and AD patients.
```{r}
names <- row.names(Means)
FC_females <- Means %>% mutate(FC_fem_ctrl_AD = 
                                 Females_AD1_Mean/Females_control1_Mean)
str(FC_females)
row.names(FC_females) <- names
```

```{r}
names <- row.names(FC_females)
FC_both <- FC_females %>% mutate(FC_male_ctrl_AD = 
                                   males_AD1_Mean/males_control1_Mean)
row.names(FC_both) <- names
str(FC_both)
```

```{r}
write.csv(FC_both, 'FC_both.csv', row.names=TRUE)
```


Combine the data of the samples with IDs in the alz1 table with the stats.
```{r}
both <- cbind(FC_both, alz1)

Both <- round(both, 3)

write.csv(Both, 'Both.csv', row.names=TRUE)
```


Top 100 genes Most expressed in fold change for females with AD:
```{r}
top100_fem <- Both[order(Both$FC_fem_ctrl_AD, decreasing=TRUE)[0:100],]
a <- row.names(top100_fem)
a
```

Top 100 genes most expressed in fold change for males with AD:
```{r}
top100_male <- Both[order(Both$FC_male_ctrl_AD, decreasing=TRUE)[0:100],]
b <- row.names(top100_male)
b
```


Bottom 100 genes that are least expressed in fold change in females with AD:
```{r}
bottom100_fem <- Both[order(Both$FC_fem_ctrl_AD, decreasing=FALSE)[0:100],]
c <-row.names(bottom100_fem)
c
```


Bottom 100 genes that are least expressed in fold change in males with AD:
```{r}
bottom100_male <- Both[order(Both$FC_male_ctrl_AD, decreasing=FALSE)[0:100],]
d <- row.names(bottom100_male)
d
```

```{r}
ab <- c(a,b)
topBoth <- unique(ab)

cd <- c(c,d)
bottomBoth <- unique(cd)
```


The unique top genes are:
```{r}
topBoth
```

The unique bottom genes are:
```{r}
bottomBoth
```

The top genes in common are those genes most expressed in fold change in females AND males who have Alzheimer:
```{r}
A <- as.data.frame(a)
B <- as.data.frame(b)

topBoth <- merge(A,B,by.x='a', by.y='b')
topBoth$a
```

The bottom genes in common are the least expressed genes in females AND males who have Alzheimer:
```{r}
C <- as.data.frame(c)
D <- as.data.frame(d)

bottomBoth <- merge(C,D, by.x='c', by.y='d')
bottomBoth$c
```


Lets look at three top expressed and three least expressed genes in common for females and males:
```{r}
b3 <- as.character(bottomBoth$c[1:3])
t3 <- as.character(topBoth$a[1:3])

mix <- c(b3,t3)
mix
```

```{r}
Mix <- as.data.frame(mix)
mBoth <- Both
mBoth$gene <- row.names(mBoth)
MixBoth <- merge(Mix, mBoth, by.x='mix', by.y='gene')

toPlot <- MixBoth[,c(1,6,7)]
colnames(toPlot)[1] <- 'gene'
toPlot
```

```{r, message=F, error=F, warning=F}
library(ggplot2)
library(tidyr)
```

```{r}

to_plot <- gather(toPlot, 'group','foldChange',2:3)
to_plot

```


```{r, message=FALSE, error=FALSE, warning=FALSE}

ggplot(data = to_plot, aes(x=gene, y=foldChange, fill=group)) +
  geom_bar(stat='identity', position=position_dodge())+
  scale_y_continuous(breaks = seq(0, 1.2, by=.1), limits=c(0,1.2))+
  scale_fill_brewer(palette='Paired') + 
  ylab('Fold Change in Alzheimer male and female Samples')


```


Some other data sets on diseases would be interesting to compare to this data on Alzheimer patients. The other data sets to compare this blood tissue type of microarray gene expression profiles are the:
hemochromatosis (iron toxicity) from https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE121620, 
epigallocatechin EGCG (green tea extract) use by overweight females in 40s found at https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE74560, 
myocardial infarction MI (heart disease) in Russian males in 40s found at https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE141512, 
antibiotic treatment in flu vaccinated patients of males and females between 18-45 years of age found at https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE120717 and https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE120719, and an 
immunization blood sample data set that only has healthy donors blood six days after receiving the tetanis-diphtheria toxoids and acellular pertussis found at https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE59697. 

Those data sets are needed for their stats with Fold Change values for each study respectively, except the tetanis shot because it is only healthy donor samples analyzing the tetanis-diptheria toxoids as an antigen to test the B lymphocytes route in the plasma cells. 

The females in the overweight study, the females in the flu study, and the females in the iron toxicity study could be cross compared to the females in this study on Alzheimer female patients. The males in the heart disease study, iron toxicity study, and flu study could be compared to the males in this Alzheimer study. And both genders from those studies could also be compared to both genders in this study. 

The plan is to get the genes these studies all have in common and compare between the genders on fold change stats of those genes. That being iron toxicity fold change, and flu vaccinated fold change compared to added antibiotics (not compared to healthy non vaccinated patients) for comparison with Alzheimer patients. Overweight females using EGCG for weight loss is exclusive to the female comparisons, and the heart disease study was exclusive to male patients.

This study's data on fold change, means, and sample values is the **both.csv** file. The flu study had days 0,3,7, and 21 comparisons and both genders in the **FoldChange_All.csv** file for that study. The iron toxicity fold change stats is in the **all-fc-grops.csv** file for the hemochromatosis study on both genders. The overweight females using EGCG to diet with the included fold change values for those using EGCG or EGCG+vitamin C+fish oil is in the **foldChange_EGCG.csv** file. And the males who have heart disease of type MI is in the file **foldChange_MI_males.csv**. The tetanis immunization samples don't have the fold change because they are all healthy donors with no age data or gender data and taken 6 days after the tetanis shot, but it is useful to compare genes with healthy females and males in the healthy and control groups **tetanisImmunizationBlood.csv** is that file.

Lets read in those files with their respective and appropriately described names.
```{r}
Alzheimer <- read.csv('both.csv', sep=',', header=T, na.strings=c('',' '))
Heart_Disease_Males <- read.csv('foldChange_MI_males.csv', sep=',',
                                header=T, na.strings=c('',' '))
TetanisImmunity <- read.csv('tetanisImmunizationBlood.csv', sep=',',
                            header=T, na.strings=c('',' '))
Overweight_Females_EGCG <- read.csv('foldChange_EGCG.csv', sep=',',
                                    header=T, na.strings=c('',' '))
flu_Vaccinated <- read.csv('FoldChange_All.csv', sep=',', header=T,
                           na.strings=c('',' '))
iron_toxicity <- read.csv('all-fc-grops.csv', sep=',', header=T, 
                          na.strings=c('', ' '))

```

Look at the column names and select the ones needed or that have fold change values to cross compare, or to compare means across samples as with the tetanis data set.

The Alzheimer column names are:
```{r}
colnames(Alzheimer)
```

The heart disease column names are:
```{r}
colnames(Heart_Disease_Males)
```

The tetanis immunization after six days taken from their blood has the following column names:
```{r}

colnames(TetanisImmunity)
```

The EGCG overweight females column names are:
```{r}
colnames(Overweight_Females_EGCG)
```

The flu vaccinated (treated with antibiotics treatment or not as 't' or 'nt' respectively) for each of the initial day 0, day 3, day 7, and day 21 profiles from three treated and three not treated samples have the following column names:
```{r}
colnames(flu_Vaccinated)
```

The iron toxicity study put the males and females into three groups of the genotypes associated with hemochromatosis or iron toxicity. It also included healthy males and females to compare to. The fold change values are for the groups each compared to the healthy groups by gender by all three groups 'overall' and by each of the 3 groups separately. The iron toxicity column names are:
```{r}
colnames(iron_toxicity)
```


In the interest of being interesting, these tables have all the original samples in each respective table for that disease group. There are some healthy samples in some such as healthy overweight females before taking EGCG, non-Alzheimer's males and females, healthy Russian males without heart disease, and healthy females and males who do not have iron toxicity relative to the iron toxicity study. There are some studies that do not have healthy controls nor separation by gender such as the tetanis immunity and flu vaccinated samples.

Since the tetanis immunized samples don't have a control or healthy group to compare to which haven't been immunized, let us compare it to the samples of healthy overweight females, healthy Russian males, healthy females and males without iron toxicity, and flu vaccinated not treated with antibiotics and flu vaccinated treated with antibiotics. The values in the tetanis immunized samples have a very high range for some genes and doesn't get lower than 0.08 for any gene. We will compare the fields of each table that has the __mean__ values across sample for each __gene__ in that group by creating a table with those means.
```{r}
tet_mean <- TetanisImmunity[,c(2,9)] #gene and tetanis immunized mean
iron_mean <- iron_toxicity[,c(1,8:11)]#gene, healthy female, healthy male
# iron toxic females, iron toxic males
flu_7day <- flu_Vaccinated[,c(1,22,17)] #gene, not treated, treated 
EGCG_females <- Overweight_Females_EGCG[,c(1,6:8)]#gene, pre,post EGCG,
# post EGCG+vitamin C+Fish Oil
MI_healthy_mean <- Heart_Disease_Males[,c(1,4,5)]#gene, healthy males, 
# MI males
alz_means <- Alzheimer[,c(1:5)]#gene, female AD, female control, male AD,
# male control
```

Let us first compare the tetanis immunized to the healthy females no iron toxicity, healthy males no iron toxicity, overweight females EGCG, healthy males no MI, healthy AD females, and healthy AD males.

Gather only the healthy females and males of those studies.
```{r}
healthy_iron <- iron_mean[,c(1:3)]
Healthy_EGCG_females <- EGCG_females #all are healthy
healthy_MImales <- MI_healthy_mean[,1:2]
healthy_AD <- alz_means[,c(1,3,5)]

dim(healthy_iron)
colnames(healthy_iron)[2:3] <- gsub('healthy','iron_healthy_',colnames(healthy_iron)[2:3])
colnames(healthy_iron)

dim(Healthy_EGCG_females)
colnames(Healthy_EGCG_females)[2:4] <- paste('overweight_females_',colnames(Healthy_EGCG_females)[2:4],sep='')
colnames(Healthy_EGCG_females)

dim(healthy_MImales)
colnames(healthy_MImales)[2] <- paste('MI_study_',colnames(healthy_MImales)[2], sep='')
colnames(healthy_MImales)

dim(healthy_AD)
colnames(healthy_AD)[2:3] <- gsub('_control1_','_no_AD_',colnames(healthy_AD)[2:3])
colnames(healthy_AD)

dim(tet_mean)
colnames(tet_mean)
```

The above tables all have a differing amount of genes in each table, so we need to merge the tables by genes in common to compare. The gene field is the 'X' field in these tables listed immediately above.We need to merge them all to the tetanis table, tet_mean, to compare with tetanis immunity gene expression values over 6 days. 
```{r}
healthy0 <- merge(tet_mean, healthy_iron, by.x='GENE_SYMBOL', by.y='X')
dim(healthy0)

healthy1 <- merge(healthy0, Healthy_EGCG_females, by.x='GENE_SYMBOL', by.y='X')
dim(healthy1)

healthy2 <- merge(healthy1, healthy_MImales, by.x='GENE_SYMBOL', by.y='X')
dim(healthy2)

healthy3 <- merge(healthy2, healthy_AD, by.x='GENE_SYMBOL', by.y='X')
dim(healthy3)

colnames(healthy3)
```

```{r}
head(healthy3)

```

We need to use dplyr to analyze this data as is for the healthy samples compared to the tetanis samples.
```{r, error=FALSE, message=FALSE, warning=FALSE}
library(dplyr)
```

Get the fold change 'FC' values for the between samples of genes using the healthy3 table.
```{r}
FC0 <- healthy3 %>% mutate(FC_iron_fem = Tetanis_Means/iron_healthy_Female_Mean)

FC1 <- FC0 %>% mutate(FC_iron_mal = Tetanis_Means/iron_healthy_Male_Mean)

FC2 <- FC1 %>% mutate(FC_heavy_fems_no_EGCG = Tetanis_Means/overweight_females_Pre_Means)

FC3 <- FC2 %>% mutate(FC_heavy_fems_EGCG = Tetanis_Means/overweight_females_Post_EGCG_Means)

FC4 <- FC3 %>% mutate(FC_heavy_fems_EGCG_plus = Tetanis_Means/overweight_females_Post_EGCG_Quercentin_Means)

FC5 <- FC4 %>% mutate(FC_healthyMI_males = Tetanis_Means/MI_study_HealthyMale_Means)

FC6 <- FC5 %>% mutate(FC_fems_noAD = Tetanis_Means/Females_no_AD_Mean)
FC7 <- FC6 %>% mutate(FC_mals_noAD = Tetanis_Means/males_no_AD_Mean)

healthy_tetanis_FCs <- FC7
dim(healthy_tetanis_FCs)
colnames(healthy_tetanis_FCs)
```

There are 13,193 genes to compare fold change values of tetanis immunized samples to the healthy control samples from the combined studies.
```{r}
write.csv(healthy_tetanis_FCs,'healthy_tetanis_FCs.csv', row.names=FALSE)
```



Now, compare the tetanis immunized to the flu vaccinated samples treated and not treated with antibiotics. The flu gene samples seemed to have the most changes in values after 7 days of receiving the flu immunization and the antibiotic or no antibiotic addition. This is why the flu immunizations for day 7 means were selected. Only the tet_mean and flu_7day tables will be used for genes in common as neither has the gender information attached.
```{r}
dim(tet_mean);colnames(tet_mean)
dim(flu_7day);colnames(flu_7day)

```

Merge these two tables of the tetanis and flu not treated (NT7 prefix) or treated (T7 prefix) means.
```{r}
tet_flu_immune <- merge(tet_mean, flu_7day, by.x='GENE_SYMBOL', by.y='Gene')
dim(tet_flu_immune)
colnames(tet_flu_immune)
```

Now for the fold change values between the Tetanis_Means and the flu immunization means with/without added antibiotics treatment, dplyr will be used to get that information.
```{r}
tetanis_flu_FCs <- tet_flu_immune %>% mutate(FC_tet_flu_nt7 = Tetanis_Means/NT7_Mean)

tetanis_flu_FCs1 <- tetanis_flu_FCs %>% mutate(FC_tet_flu_t7 = Tetanis_Means/T7_Mean)

dim(tetanis_flu_FCs1)
colnames(tetanis_flu_FCs1)
head(tetanis_flu_FCs1)
```

```{r}
write.csv(tetanis_flu_FCs1, 'tetanis_flu_FCs.csv', row.names=FALSE)
```





