---
title: "Alzheimers Disease Brain Blood Samples"
author: "Janis Corona"
date: "1/17/2020"
output:
  html_document:
    df_print: paged
  pdf_document: default
  word_document: default
---

This analysis is done using the NCBI gene expression samples taken from a study that used the middle temporal gyrus side of the brain of frozen samples taken from Alzheimer's patients that totalled 78 samples of control and repeat microarray gene expression results. The file was able to be downloaded and unzipped complete with the platform gene symbol ID already attached. A separate file of the meta information that includes the age, gender, tissue type and disease as Alzheimer's Disease (AD) or control was made from the Series information for each sample. The age range for these samples of healthy and AD patients is from 70-95 years of age.

This study can be linked to with all the sample and meta information at: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE109887
The link to the 41 mb text file of series of samples with ID that is too large for github, or my github account type is: ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE109nnn/GSE109887/matrix/GSE109887_series_matrix.txt.gz 

Open the original Sample ID values text file and write out as a csv file.
```{r}
alz <- read.delim('GSE109887_series_matrix.txt', sep='\t', header=TRUE, na.strings=c('',' '), comment.char='!')

write.csv(alz,'alzheimerSamples.csv', row.names=FALSE)

```

```{r}
head(alz)
```


Lets read in the meta information for age, gender, and disease type as AD or control.
```{r}
meta <- read.csv('AlzheimerAgeGenderTissueSamplesMeta.csv', sep=',',
                 header=TRUE, na.strings=c('',' '))
```

```{r}
head(meta)
```


Transpose the meta to fields instead of rows
```{r}

meta1 <- as.data.frame(t(meta))
names <- as.character(meta$sampleID)
colnames(meta1) <- names
meta2 <- meta1[-1,]
head(meta2)
```


Separate the data into gender and control or AD data sets respectively in each gender and for both genders.

Data tables of males and females for meta only:
```{r}
row.names(alz) <- alz$ID_REF
alz1 <- alz[,-1]

row.names(meta) <- meta$sampleID
Meta <- meta[,-1]

names <- row.names(Meta)
Meta1 <- as.data.frame(t(Meta))
colnames(Meta1) <- names

fem <- grep('F', Meta1$gender)
mal <- grep('M', Meta1$gender)

Fem <- Meta1[fem,]
Mal <- Meta1[mal,]

Fem$sampleID <- as.factor(row.names(Fem))
Mal$sampleID <- as.factor(row.names(Mal))
```

Display the fields of these gender specific tables by meta information:
```{r}
colnames(Fem)
colnames(Mal)
```

The same indices of each row name is the same as the alz1, alzheimer table colnames indices. So, use those same female and male indices values from the regex commands to separate the alzheimer table into male and female tables.
```{r}
Females <- alz1[,fem]
Males <- alz1[,mal]
```



Data tables of control or AD for males and for females for meta only:
```{r}
femControl <- grep('control', Fem$disease)
malControl <- grep('control', Mal$disease)

femAD <- grep('AD', Fem$disease)
malAD <- grep('AD', Mal$disease)

FemCtrl <- Fem[femControl,]
MalCtrl <- Mal[malControl,]

FemAD <- Fem[femAD,]
MalAD <- Mal[malAD,]
```

Display the AD or Control tables by gender for meta information:
```{r}
colnames(FemCtrl)
colnames(FemAD)
colnames(FemAD)
colnames(MalAD)
```

Now use that same index information within each gender table derived from the alzheimer gene expression data:
```{r}
Females_AD <- Females[,femAD]
Females_control <- Females[,femControl]

males_AD <- Males[,malAD]
males_control <- Males[, malControl]
```

Write these last tables to csv file:
```{r}
write.csv(Females_AD, 'females_Alzheimers.csv', row.names=TRUE)
write.csv(Females_control, 'females_healthy_control.csv', row.names=TRUE)
write.csv(males_AD, 'males_Alzheimers.csv', row.names=TRUE)
write.csv(males_control, 'males_healthy_control.csv', row.names=TRUE)
```

What is the age range on these samples of healthy controls and AD patients?
```{r}
age1 <- range(as.numeric(as.character(FemAD$age)))
age2 <- range(as.numeric(as.character(FemCtrl$age)))
age3 <- range(as.numeric(as.character(MalAD$age)))
age4 <- range(as.numeric(as.character(MalCtrl$age)))

char <- c('The minimum age: ', 'The maximum age: ')

AD_fem <- paste0(char,age1, sep='')
AD_mal <- paste0(char,age3, sep='')
ctrl_fem <- paste0(char,age2, sep='')
ctrl_mal <- paste0(char, age4, sep='')

adf <- c('Alzheimer females: ')
adm <- c('Alzheimer males: ')
cf <- c('healthy females: ')
cm <- c('healthy males: ')

ranges <- c(adf,AD_fem, adm, AD_mal, cf, ctrl_fem, cm, ctrl_mal)
ranges
```

The age range in all of this data for these healthy and AD patients is age 70 to age 95 years of age.Just useful to know when cross comparing blood microarray samples across the females from our overweight females using epigallocatechin (EGCG) in their 40s, the males from Russia who are also in their 40s with or without myocardial infarction (MI), the hemochromatosis (iron toxicity) males and females averaging their 40s in years of age with or without iron toxicity, and the flu vaccinated blood samples of males and females also in their 40s with or without antibiotic treatment combined with flu vaccination. 


Lets do some data analysis of the means and fold change between the groups to create some additional statistical information on these Alzheimers blood samples and healthy controls of elderly patients. We will use dplyr for this and plot with ggplot2 later.

```{r, error=FALSE, message=F, warning=F}
library(dplyr)
```

First lets look at the control and the AD tables within the gender specific tables: Females_AD, Females_control, males_AD, males_control. Get the means of each gene and compare into one table. This and these tables of data were already cleaned to only have one gene per row, due to not having to merge the platform fields to the series table and duplicate genes for missing data in the additional rows like previous studies mentioned earlier required.

Get the row means of each table and add to each table.
```{r}
names <- row.names(Females_AD)
Females_AD1 <- rowMeans(Females_AD)
#colnames(Females_AD1) <- 'fem_AD_Mean'

Females_control1 <- rowMeans(Females_control)
#colnames(Females_control1) <- 'fem_ctrl_Mean'

males_AD1 <- rowMeans(males_AD)
#colnames(males_AD1) <- 'mal_AD_Mean'

males_control1 <- rowMeans(males_control)
#colnames(males_control1) <- 'mal_ctrl_Mean'
```

```{r}
means <- cbind(Females_AD1, Females_control1, males_AD1, males_control1)
Means <- as.data.frame(means)
colnames(Means) <- paste(colnames(Means),'_Mean',sep='')
str(Means)
```

Now for the fold change values between the Means of the female control and AD patients, then for the male control and AD patients.
```{r}
names <- row.names(Means)
FC_females <- Means %>% mutate(FC_fem_ctrl_AD = 
                                 Females_AD1_Mean/Females_control1_Mean)
str(FC_females)
row.names(FC_females) <- names
```

```{r}
names <- row.names(FC_females)
FC_both <- FC_females %>% mutate(FC_male_ctrl_AD = 
                                   males_AD1_Mean/males_control1_Mean)
row.names(FC_both) <- names
str(FC_both)
```

```{r}
write.csv(FC_both, 'FC_both.csv', row.names=TRUE)
```


Combine the data of the samples with IDs in the alz1 table with the stats.
```{r}
both <- cbind(FC_both, alz1)

Both <- round(both, 3)

write.csv(Both, 'Both.csv', row.names=TRUE)
```


Top 100 genes Most expressed in fold change for females with AD:
```{r}
top100_fem <- Both[order(Both$FC_fem_ctrl_AD, decreasing=TRUE)[0:100],]
a <- row.names(top100_fem)
a
```

Top 100 genes most expressed in fold change for males with AD:
```{r}
top100_male <- Both[order(Both$FC_male_ctrl_AD, decreasing=TRUE)[0:100],]
b <- row.names(top100_male)
b
```


Bottom 100 genes that are least expressed in fold change in females with AD:
```{r}
bottom100_fem <- Both[order(Both$FC_fem_ctrl_AD, decreasing=FALSE)[0:100],]
c <-row.names(bottom100_fem)
c
```


Bottom 100 genes that are least expressed in fold change in males with AD:
```{r}
bottom100_male <- Both[order(Both$FC_male_ctrl_AD, decreasing=FALSE)[0:100],]
d <- row.names(bottom100_male)
d
```

```{r}
ab <- c(a,b)
topBoth <- unique(ab)

cd <- c(c,d)
bottomBoth <- unique(cd)
```


The unique top genes are:
```{r}
topBoth
```

The unique bottom genes are:
```{r}
bottomBoth
```

The top genes in common are those genes most expressed in fold change in females AND males who have Alzheimer:
```{r}
A <- as.data.frame(a)
B <- as.data.frame(b)

topBoth <- merge(A,B,by.x='a', by.y='b')
topBoth$a
```

The bottom genes in common are the least expressed genes in females AND males who have Alzheimer:
```{r}
C <- as.data.frame(c)
D <- as.data.frame(d)

bottomBoth <- merge(C,D, by.x='c', by.y='d')
bottomBoth$c
```


Lets look at three top expressed and three least expressed genes in common for females and males:
```{r}
b3 <- as.character(bottomBoth$c[1:3])
t3 <- as.character(topBoth$a[1:3])

mix <- c(b3,t3)
mix
```

```{r}
Mix <- as.data.frame(mix)
mBoth <- Both
mBoth$gene <- row.names(mBoth)
MixBoth <- merge(Mix, mBoth, by.x='mix', by.y='gene')

toPlot <- MixBoth[,c(1,6,7)]
colnames(toPlot)[1] <- 'gene'
toPlot
```

```{r, message=F, error=F, warning=F}
library(ggplot2)
library(tidyr)
```

```{r}

to_plot <- gather(toPlot, 'group','foldChange',2:3)
to_plot

```


```{r, fig.width=7, fig.height=7}

ggplot(data = to_plot, aes(x=gene, y=foldChange, fill=group)) +
  geom_bar(stat='identity', position=position_dodge())+
  scale_y_continuous(breaks = seq(0, 1.2, by=.1), limits=c(0,1.2))+
  scale_fill_brewer(palette='Paired') + 
  ggtitle('Alzheimer\'s Disease for Males and Females')+
  ylab('Fold Change')+
  xlab('Three Most and Least Expressed Genes')
```


***

Some other data sets on diseases would be interesting to compare to this data on Alzheimer patients. The other data sets to compare this blood tissue type of microarray gene expression profiles are the:
hemochromatosis (iron toxicity) from https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE121620, 
epigallocatechin EGCG (green tea extract) use by overweight females in 40s found at https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE74560, 
myocardial infarction MI (heart disease) in Russian males in 40s found at https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE141512, 
antibiotic treatment in flu vaccinated patients of males and females between 18-45 years of age found at https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE120717 and https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE120719, and an 
immunization blood sample data set that only has healthy donors blood six days after receiving the tetanis-diphtheria toxoids and acellular pertussis found at https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE59697. 

Those data sets are needed for their stats with Fold Change values for each study respectively, except the tetanis shot because it is only healthy donor samples analyzing the tetanis-diptheria toxoids as an antigen to test the B lymphocytes route in the plasma cells. 

The females in the overweight study, the females in the flu study, and the females in the iron toxicity study could be cross compared to the females in this study on Alzheimer female patients. The males in the heart disease study, iron toxicity study, and flu study could be compared to the males in this Alzheimer study. And both genders from those studies could also be compared to both genders in this study. 

The plan is to get the genes these studies all have in common and compare between the genders on fold change stats of those genes. That being iron toxicity fold change, and flu vaccinated fold change compared to added antibiotics (not compared to healthy non vaccinated patients) for comparison with Alzheimer patients. Overweight females using EGCG for weight loss is exclusive to the female comparisons, and the heart disease study was exclusive to male patients.

This study's data on fold change, means, and sample values is the **both.csv** file. The flu study had days 0,3,7, and 21 comparisons and both genders in the **FoldChange_All.csv** file for that study. The iron toxicity fold change stats is in the **all-fc-grops.csv** file for the hemochromatosis study on both genders. The overweight females using EGCG to diet with the included fold change values for those using EGCG or EGCG+vitamin C+fish oil is in the **foldChange_EGCG.csv** file. And the males who have heart disease of type MI is in the file **foldChange_MI_males.csv**. The tetanis immunization samples don't have the fold change because they are all healthy donors with no age data or gender data and taken 6 days after the tetanis shot, but it is useful to compare genes with healthy females and males in the healthy and control groups **tetanisImmunizationBlood.csv** is that file.

***

Lets read in those files with their respective and appropriately described names.
```{r}
Alzheimer <- read.csv('both.csv', sep=',', header=T, na.strings=c('',' '))
Heart_Disease_Males <- read.csv('foldChange_MI_males.csv', sep=',',
                                header=T, na.strings=c('',' '))
TetanisImmunity <- read.csv('tetanisImmunizationBlood.csv', sep=',',
                            header=T, na.strings=c('',' '))
Overweight_Females_EGCG <- read.csv('foldChange_EGCG.csv', sep=',',
                                    header=T, na.strings=c('',' '))
flu_Vaccinated <- read.csv('FoldChange_All.csv', sep=',', header=T,
                           na.strings=c('',' '))
iron_toxicity <- read.csv('all-fc-grops.csv', sep=',', header=T, 
                          na.strings=c('', ' '))

```

Look at the column names and select the ones needed or that have fold change values to cross compare, or to compare means across samples as with the tetanis data set.

The Alzheimer column names are:
```{r}
Alzheimer1 <- Alzheimer
colnames(Alzheimer1)[c(2:85)] <- paste('Alz_',
                                       colnames(Alzheimer1)[c(2:85)],
                                      sep='')
colnames(Alzheimer1)
```

The heart disease column names are:
```{r}
Heart_Disease_Males1 <- Heart_Disease_Males
colnames(Heart_Disease_Males1)[c(4,6:11)] <- gsub('Male', 'MI_Male',
                      colnames(Heart_Disease_Males1)[c(4,6:11)])
Heart_Disease_Males1 <- Heart_Disease_Males1[,-1]
colnames(Heart_Disease_Males1)

```

The tetanis immunization after six days taken from their blood has the following column names:
```{r}

colnames(TetanisImmunity)
min(TetanisImmunity[,3:9])
max(TetanisImmunity[,3:9])
```

Take the log2 scale of the tetanis data set to make the values the same scale as the flu vaccinated and other samples that used log2 scale.
```{r}
TetanisImmunity1 <- log2(TetanisImmunity[,3:9]+1)#avoid negative values
TetanisImmunity1$GENE_SYMBOL <- TetanisImmunity$GENE_SYMBOL
min(TetanisImmunity1[,1:6])
max(TetanisImmunity1[,1:6])

Tetanis_Immunity1 <- TetanisImmunity1[,c(8,7,1:6)]
colnames(Tetanis_Immunity1)[3:8] <- paste('tetanis_',
                    colnames(Tetanis_Immunity1)[3:8], sep='')
colnames(Tetanis_Immunity1)
```

The EGCG overweight females column names are:
```{r}
Overweight_Females_EGCG1 <- Overweight_Females_EGCG
colnames(Overweight_Females_EGCG1)[9:36] <- paste('EGCG',                                                  colnames(Overweight_Females_EGCG1[9:36]),
                                                  sep='_')
colnames(Overweight_Females_EGCG1)
```

The flu vaccinated (treated with antibiotics treatment or not as 't' or 'nt' respectively) for each of the initial day 0, day 3, day 7, and day 21 profiles from three treated and three not treated samples have the following column names:
```{r}
flu_Vaccinated1 <- flu_Vaccinated
colnames(flu_Vaccinated1)[24:53] <- paste('flu',colnames(flu_Vaccinated1)[24:53],
                                     sep='_')
colnames(flu_Vaccinated1)
```

The iron toxicity study put the males and females into three groups of the genotypes associated with hemochromatosis or iron toxicity. It also included healthy males and females to compare to. The fold change values are for the groups each compared to the healthy groups by gender by all three groups 'overall' and by each of the 3 groups separately. The iron toxicity column names are:
```{r}
iron_toxicity1 <- iron_toxicity
colnames(iron_toxicity1)[c(2:9,12:43)] <- paste('hemo_',colnames(iron_toxicity1)[c(2:9,12:43)], 
                                 sep='')
colnames(iron_toxicity1)[10:11] <- gsub('hemo','hemo_',colnames(iron_toxicity1)[10:11])

colnames(iron_toxicity1)
```

***
The above tables that were just made that added the type of sample study to the NCBI GEO accession ID's and other fields missing table identifier type are now going to be merged and written to csv to use in other programs or purposes. It will also be added to Kaggle datasets.

```{r}
K0 <- merge(Alzheimer1, Heart_Disease_Males1, by.x='X', by.y='Symbol')
K1 <- merge(Tetanis_Immunity1, K0, by.x='GENE_SYMBOL', by.y='X')
K2 <- merge(K1, flu_Vaccinated1, by.x='GENE_SYMBOL', by.y='Gene')
K3 <- merge(K2, Overweight_Females_EGCG1, by.x='GENE_SYMBOL', by.y='X')
K4 <- merge(K3, iron_toxicity1, by.x='GENE_SYMBOL', by.y='X')

GSM <- grep('GSM',colnames(K4))
all6samples <- K4[,GSM]
all6stats <- K4[,-GSM]
all6 <- cbind(all6stats,all6samples)
colnames(all6)

write.csv(all6, 'all_6_studies.csv', row.names=FALSE)
```


***

In the interest of being interesting, these tables have all the original samples in each respective table for that disease group. There are some healthy samples in some such as healthy overweight females before taking EGCG, non-Alzheimer's males and females, healthy Russian males without heart disease, and healthy females and males who do not have iron toxicity relative to the iron toxicity study. There are some studies that do not have healthy controls nor separation by gender such as the tetanis immunity and flu vaccinated samples.

Since the tetanis immunized samples don't have a control or healthy group to compare to which haven't been immunized, let us compare it to the samples of healthy overweight females, healthy Russian males, healthy females and males without iron toxicity, and flu vaccinated not treated with antibiotics and flu vaccinated treated with antibiotics. The values in the tetanis immunized samples have a very high range for some genes and doesn't get lower than 0.08 for any gene. We will compare the fields of each table that has the __mean__ values across sample for each __gene__ in that group by creating a table with those means.
```{r}
tet_mean <- TetanisImmunity1[,c(7,8)] #gene and tetanis immunized mean
iron_mean <- iron_toxicity[,c(1,8:11)]#gene, healthy female, healthy male
# iron toxic females, iron toxic males
flu_7day <- flu_Vaccinated[,c(1,22,17)] #gene, not treated, treated 
EGCG_females <- Overweight_Females_EGCG[,c(1,6:8)]#gene, pre,post EGCG,
# post EGCG+vitamin C+Fish Oil
MI_healthy_mean <- Heart_Disease_Males[,c(1,4,5)]#gene, healthy males, 
# MI males
alz_means <- Alzheimer[,c(1:5)]#gene, female AD, female control, male AD,
# male control
```

Let us first compare the tetanis immunized to the healthy females no iron toxicity, healthy males no iron toxicity, overweight females EGCG, healthy males no MI, healthy AD females, and healthy AD males.

Gather only the healthy females and males of those studies.
```{r}
healthy_iron <- iron_mean[,c(1:3)]
Healthy_EGCG_females <- EGCG_females #all are healthy
healthy_MImales <- MI_healthy_mean[,1:2]
healthy_AD <- alz_means[,c(1,3,5)]

dim(healthy_iron)
colnames(healthy_iron)[2:3] <- gsub('healthy','iron_healthy_',colnames(healthy_iron)[2:3])
colnames(healthy_iron)

dim(Healthy_EGCG_females)
colnames(Healthy_EGCG_females)[2:4] <- paste('overweight_females_',colnames(Healthy_EGCG_females)[2:4],sep='')
colnames(Healthy_EGCG_females)

dim(healthy_MImales)
colnames(healthy_MImales)[2] <- paste('MI_study_',colnames(healthy_MImales)[2], sep='')
colnames(healthy_MImales)

dim(healthy_AD)
colnames(healthy_AD)[2:3] <- gsub('_control1_','_no_AD_',colnames(healthy_AD)[2:3])
colnames(healthy_AD)

dim(tet_mean)
colnames(tet_mean)
```

The above tables all have a differing amount of genes in each table, so we need to merge the tables by genes in common to compare. The gene field is the 'X' field in these tables listed immediately above.We need to merge them all to the tetanis table, tet_mean, to compare with tetanis immunity gene expression values over 6 days. 
```{r}
healthy0 <- merge(tet_mean, healthy_iron, by.x='GENE_SYMBOL', by.y='X')
dim(healthy0)

healthy1 <- merge(healthy0, Healthy_EGCG_females, by.x='GENE_SYMBOL', by.y='X')
dim(healthy1)

healthy2 <- merge(healthy1, healthy_MImales, by.x='GENE_SYMBOL', by.y='X')
dim(healthy2)

healthy3 <- merge(healthy2, healthy_AD, by.x='GENE_SYMBOL', by.y='X')
dim(healthy3)

colnames(healthy3)
```

```{r}
head(healthy3)

```

We need to use dplyr to analyze this data as is for the healthy samples compared to the tetanis samples.
```{r, error=FALSE, message=FALSE, warning=FALSE}
library(dplyr)
```

Get the fold change 'FC' values for the between samples of genes using the healthy3 table.
```{r}
FC0 <- healthy3 %>% mutate(FC_iron_fem = Tetanis_Means/iron_healthy_Female_Mean)

FC1 <- FC0 %>% mutate(FC_iron_mal = Tetanis_Means/iron_healthy_Male_Mean)

FC2 <- FC1 %>% mutate(FC_heavy_fems_no_EGCG = Tetanis_Means/overweight_females_Pre_Means)

FC3 <- FC2 %>% mutate(FC_heavy_fems_EGCG = Tetanis_Means/overweight_females_Post_EGCG_Means)

FC4 <- FC3 %>% mutate(FC_heavy_fems_EGCG_plus = Tetanis_Means/overweight_females_Post_EGCG_Quercentin_Means)

FC5 <- FC4 %>% mutate(FC_healthyMI_males = Tetanis_Means/MI_study_HealthyMale_Means)

FC6 <- FC5 %>% mutate(FC_fems_noAD = Tetanis_Means/Females_no_AD_Mean)
FC7 <- FC6 %>% mutate(FC_mals_noAD = Tetanis_Means/males_no_AD_Mean)

healthy_tetanis_FCs <- FC7
dim(healthy_tetanis_FCs)
colnames(healthy_tetanis_FCs)
```


There are 13,193 genes to compare fold change values of tetanis immunized samples to the healthy control samples from the combined studies.
```{r}
write.csv(healthy_tetanis_FCs,'healthy_tetanis_FCs.csv', row.names=FALSE)
```



Now, compare the tetanis immunized to the flu vaccinated samples treated and not treated with antibiotics. The flu gene samples seemed to have the most changes in values after 7 days of receiving the flu immunization and the antibiotic or no antibiotic addition. This is why the flu immunizations for day 7 means were selected. Only the tet_mean and flu_7day tables will be used for genes in common as neither has the gender information attached.
```{r}
dim(tet_mean);colnames(tet_mean)
dim(flu_7day);colnames(flu_7day)

```

Merge these two tables of the tetanis and flu not treated (NT7 prefix) or treated (T7 prefix) means.
```{r}
tet_flu_immune <- merge(tet_mean, flu_7day, by.x='GENE_SYMBOL', by.y='Gene')
dim(tet_flu_immune)
colnames(tet_flu_immune)
```

Now for the fold change values between the Tetanis_Means and the flu immunization means with/without added antibiotics treatment, dplyr will be used to get that information.
```{r}
tetanis_flu_FCs <- tet_flu_immune %>% mutate(FC_tet_flu_nt7 = Tetanis_Means/NT7_Mean)

tetanis_flu_FCs1 <- tetanis_flu_FCs %>% mutate(FC_tet_flu_t7 = Tetanis_Means/T7_Mean)

dim(tetanis_flu_FCs1)
colnames(tetanis_flu_FCs1)
head(tetanis_flu_FCs1)
```

```{r}
write.csv(tetanis_flu_FCs1, 'tetanis_flu_FCs.csv', row.names=FALSE)
```

Going to genecards.org I searched for Immune genes and found some genes that I want to analyze in this data set, tetanis_flu_FCs1, to see how immune cells being treated with the tetanis antibodies after six days and the separate study of flu immunized cells being treated with antibiotics after seven days, and not treated with antibiotics after 7 days.These genes are:
* __CD4__ (big autoimmune indicator in lupus)
* __IL2__ (produced by Produced by T-cells in response to antigenic or mitogenic...and other activities crucial to regulation of the immune response. Can stimulate B-cells, monocytes, lymphokine-activated... (UniProtKB/Swiss-Prot))
* __IFNG__ (Produced by lymphocytes activated by specific antigens or mitogens. IFN-gamma, in addition to having antiviral activity, has important immunoregulatory functions. It is a potent activator of macrophages, it has antiproliferative effects on transformed cells and it can potentiate the antiviral and antitumor effects of the type I interferons.)
* __IL10__ (Major immune regulatory cytokine that acts on many cells of the immune system where it has profound anti-inflammatory functions, limiting excessive tissue disruption caused by inflammation.)
* __IL6__ (Cytokine with a wide variety of biological functions. It is a potent inducer of the acute phase response. Plays an essential role in the final differentiation of B-cells into Ig-secreting cells Involved in lymphocyte and monocyte differentiation. Acts on B-cells, T-cells, hepatocytes, hematopoietic progenitor cells and cells of the CNS. Required for the generation of T(H)17 cells. Also acts as a myokine. It is discharged into the bloodstream after muscle contraction and acts to increase the breakdown of fats and to improve insulin resistance. It induces myeloma and plasmacytoma growth and induces nerve cells differentiation.)

```{r}
CD4 <- grep('^CD4$', tetanis_flu_FCs1$GENE_SYMBOL)
IL2 <- grep('^IL2$', tetanis_flu_FCs1$GENE_SYMBOL)
IFNG <- grep('^IFNG$', tetanis_flu_FCs1$GENE_SYMBOL)
IL10 <- grep('^IL10$', tetanis_flu_FCs1$GENE_SYMBOL)
IL6 <- grep('^IL6$', tetanis_flu_FCs1$GENE_SYMBOL)

tet_flu_immune_genes <- tetanis_flu_FCs1[c(CD4,IL2,IFNG,IL10,IL6),]
head(tet_flu_immune_genes)
```

We need to tidy this data set for plotting with the tidyr and ggplot2 libraries.
```{r, message=FALSE, error=FALSE, warning=FALSE}
library(tidyr)
library(ggplot2)

```

Tidy the table into a tibble
```{r}
tet_flu_tidy <- gather(tet_flu_immune_genes, 'group','foldChange',2:6)
tet_flu_tidy$foldChange <- round(tet_flu_tidy$foldChange,3)
head(tet_flu_tidy)
```

Plot the table or tibble of data above using bar plot to see how the gene expression values of immune genes compare.
```{r, fig.width=7, fig.height=7}
ggplot(data = tet_flu_tidy, aes(x=GENE_SYMBOL, y=foldChange, fill=group)) +
  geom_bar(stat='identity', position=position_dodge())+
  scale_y_continuous(breaks = seq(0, 30, by=2), limits=c(0,32))+
  scale_fill_brewer(palette='Paired') + 
  ggtitle('Tetanis and Flu Vaccinated Samples')+
  ylab('Fold Change')+
  xlab('Immune Gene')

```

***

Now that we  have a couple bar charts of these immune genes selected to compare across the fold change values and mean values of the tetanis and flu vaccinated samples that were and weren't treated with antibiotics, lets identify some conditions that stand out for gene expression values of the fold change values of tetanis to flu vaccinated.

It is apparantly clear that the tetnis vaccinated mean values are highest in almost every immune gene above except for the IL2 gene which is produced by the T-cells and crucial to the immune response. There is a higher fold change in tetanis to flu without antibiotics than the fold change of tetanis to flu treated with antibiotics in the CD4, IL10, and IL2 genes, but the IFNG (antiviral properties) and IL6 genes have a lower fold change with the IL6 (increases break down of fats and insulin resistance) gene dramatically decreased. 

From the above sample the IL2 gene is not visible, so we should scale it down knowing the other genes will be missing from the next plot.
```{r, message=FALSE, error=FALSE, warning=FALSE,fig.width=7, fig.height=7}
ggplot(data = tet_flu_tidy, aes(x=GENE_SYMBOL, y=foldChange, fill=group)) +
  geom_bar(stat='identity', position=position_dodge())+
  scale_y_continuous(breaks = seq(0, 4, by=.5), limits=c(0,4))+
  scale_fill_brewer(palette='Paired') + 
  ggtitle('Tetanis and Flu Vaccinated Samples for IL2 Range')+
  ylab('Fold Change')+
  xlab('Immune Gene')

```

***

What about comparing all the healthy female and male samples to the tetanis vaccinated samples and looking at those same immune genes, and doing the same for the flu samples? Would this show some interesting data? This table is the healthy_tetanis_FCs table.
```{r}
colnames(healthy_tetanis_FCs)

```

Use dplyr to get gene means of all the sample means of healthy patients except the tetanis means. This will combine the mixed genders and ages of the healthy blood samples to compare with the tetanis vaccinated samples.
```{r}
healthy <- healthy_tetanis_FCs[,c(1:10)]

CD4 <- grep('^CD4$', healthy$GENE_SYMBOL)
IL2 <- grep('^IL2$', healthy$GENE_SYMBOL)
IFNG <- grep('^IFNG$', healthy$GENE_SYMBOL)
IL10 <- grep('^IL10$', healthy$GENE_SYMBOL)
IL6 <- grep('^IL6$', healthy$GENE_SYMBOL)

healthy_ig <- healthy[c(CD4,IL2,IFNG,IL10,IL6),]

healthy_ig$all_healthy_Means <- rowMeans(healthy_ig[3:10])
healthy_FC <- healthy_ig %>% mutate(FC_tet_healthy = Tetanis_Means/all_healthy_Means)

healthy_tet_fc <- healthy_FC[,c(1,2,11,12)]
```


Tidy this table, healthy_tet_fc, into a tibble to plot a bar chart of the values.
```{r}
healthy_tet_fc_tidy <- gather(healthy_tet_fc, 'group','foldChange',2:4)
healthy_tet_fc_tidy$foldChange <- round(healthy_tet_fc_tidy$foldChange,3)
head(healthy_tet_fc_tidy)


```

Now plot the results.
```{r}
ggplot(data = healthy_tet_fc_tidy, aes(x=GENE_SYMBOL, y=foldChange, fill=group)) +
  geom_bar(stat='identity', position=position_dodge())+
  scale_y_continuous(breaks = seq(0, 30, by=2), limits=c(0,32))+
  scale_fill_brewer(palette='Paired') + 
  ggtitle('Tetanis Vaccinated to Healthy Samples')+
  ylab('Fold Change')+
  xlab('Immune Gene')

```

***

From the bar chart above, the Tetanis mean values are greatest (dramatically) in all immune genes selected except for the IL2 gene, as was the same for the tetanis to flu immunized samples. The fold change for tetanis vaccinated compared to healthy non-vaccinated samples show that Tetanis fold change values are all more than a 100% increase for these specific immune genes, except for the IL2 gene that will be shown in the plot below.
```{r, warning=FALSE}
ggplot(data = healthy_tet_fc_tidy, aes(x=GENE_SYMBOL, y=foldChange, fill=group)) +
  geom_bar(stat='identity', position=position_dodge())+
  scale_y_continuous(breaks = seq(0, 3, by=.5), limits=c(0,3.5))+
  scale_fill_brewer(palette='Paired') + 
  ggtitle('Tetanis Vaccinated and Healthy Samples for IL2 Range')+
  ylab('Fold Change')+
  xlab('Immune Gene')

```

***

There are some more analytics or comparisons that can be done with this sort of data warehouse on gene expression values from blood in microarray medium. 
The flu vaccinated samples, flu_7day, could then be compared to these healthy samples in the healthy table. The EGCG effects on overweight females, the hemochromatosis patients' gene values can be compared to find how these specific immune genes compare to immunized flu and tetanis samples are a few other comparisons that can be made.
```{r}
colnames(flu_7day)[2:3] <- paste('flu_',colnames(flu_7day)[2:3], sep='')
colnames(flu_7day)
colnames(healthy)
```

```{r}
flu_healthy <- merge(flu_7day, healthy, by.x='Gene', by.y='GENE_SYMBOL')
colnames(flu_healthy)
```

```{r}
CD4 <- grep('^CD4$', flu_healthy$Gene)
IL2 <- grep('^IL2$', flu_healthy$Gene)
IFNG <- grep('^IFNG$', flu_healthy$Gene)
IL10 <- grep('^IL10$', flu_healthy$Gene)
IL6 <- grep('^IL6$', flu_healthy$Gene)

flu_healthy1 <- flu_healthy[c(CD4,IL2,IFNG,IL10,IL6),]
head(flu_healthy1)

```

These are the fold change value comparisons to healthy mixed ages and gender samples. Where NT7 is not treated with antibiotics and T7 was treated for the flu fold change values only.
```{r}
flu_healthy1$All_Healthy_Mean <- rowMeans(flu_healthy1[,5:12])
flu_healthy2 <- flu_healthy1 %>% mutate(FC_tet_healthy = 
                                          Tetanis_Means/All_Healthy_Mean)
flu_healthy3 <- flu_healthy2 %>% mutate(FC_flu_NT7_healthy = 
                                          flu_NT7_Mean/All_Healthy_Mean)
flu_healthy4 <- flu_healthy3 %>% mutate(FC_flu_T7_healthy = 
                                          flu_T7_Mean/All_Healthy_Mean)
Flu_Healthy <- flu_healthy4[,c(1:4,13:16)]
```

Use tidy on this data.
```{r}
Flu_Healthy_tidy <- gather(Flu_Healthy, 'group','foldChange',2:8)
Flu_Healthy_tidy$foldChange <- round(Flu_Healthy_tidy$foldChange,3)
head(Flu_Healthy_tidy)

```


```{r, fig.width=7, fig.height=7}
ggplot(data = Flu_Healthy_tidy, aes(x=Gene, y=foldChange, fill=group)) +
  geom_bar(stat='identity', position=position_dodge())+
  scale_y_continuous(breaks = seq(0, 27, by=2), limits=c(0,28))+
  scale_fill_brewer(palette='Paired') + 
  ggtitle('Flu Vaccinated Immune Genes Compared to Healthy')+
  ylab('Fold Change')+
  xlab('Immune Gene')

```

***

The above bar chart shows the Mean values for Tetanis and Flu vaccinated samples for these immune genes are higher than the mean values for these immune genes in non-vaccinated healthy samples of mixed genders and ages 18-95 years old. 

The fold change values compared to healthy for the flu vaccinated and treated with antibiotics show these immune genes to be higher than the treated in only the IL10 and IL2 immune genes. 

Again, IL2 is the only gene that the tetanis mean and fold change in gene expression values is not greater than the healthy mean values or the flu vaccinated mean values and fold change values compared to healthy gene expression means.

In the IL6 gene, the fold change value of the flu vaccinated not treated with antibiotics is more than double the fold change value of the flu vaccinated that was treated with antibiotics. This could mean seven days after being treated with antibiotics when also being vaccinated for the flu, that the IL6 immune gene slows down expression values below the healthy levels.

***

It still remains to compare these samples of Alzheimer's Disease patients across samples of the healthy controls, and the same for the EGCG overweight female treatment, the heart disease males, iron toxicity males and females for the vaccinations. Since immune genes are a good way of telling how stressed a person's body is outside of what he or she can control this gives another option for analyzing health of a person to get vaccinated or when testing a new diet or discovering gene targets involved in an increased risk of death in heart disease or Alzheimer's disease. 

What questions are you thinking about discovering given this data? Can you uncover what you know piled on top of what this data can explain for you to help answer the questions you have? If so, then play around with this data to answer those questions. If not, then go get more data or develop a different way to analyze this warehouse of data, then make note of any questions you have that you find interesting or curious and answer it with the data.

***

***

After letting this analysis sit for a while, lets get the data of all 236 samples of stats and study samples ready for machine learning with R and save for later using a classification variable that will give the study each sample was derived from. To do this we must first identify each sample from its respective study and retrieve the sample meta as healthy or treatment/disease type. This dataset is already in our environment from earlier if you saved your RStudio information or kept this file open the entire time while running the previous commands, all6, is the table to use. Otherwise it was written to csv as 'all_6_studies.csv' to read in.
```{r}
#Return the column names of all6
colnames(all6)
```

The first 58 fields are the stat fields created for respective data sets and won't be included in the machine learning data set. 

- The tetanis data sets didn't have any age or gender information provided and were all healthy so it will not need to be identified separately from healthy classification. 

- The next samples are for Alzheimers, and these samples don't say the healthy or disease state, so they will need to have that information attached for **columns 65 to 142**. 

- The MI samples as **columns 143 to 154** are already identified as healthy or MI.

- The flu samples are identified by flu as is with antibiotics or flu with out antibiotics from healthy donors in **columns 155 to 184**

- The EGCG samples are also identified already as using or not using EGCG in **columns 185 to 212** in the column name.

- The hemochromatosis columns in **213 to 236** need to be identified as healthy or disease.

I went through the subset tables in the Alzheimer and hemochromatosis gender and healthy tables to get the respective indices of the gender and healthy identified samples in this table.Revisiting the Flu+anti or not study, the day 0 is also the healthy state pre-vaccination that can be used as healthy blood , but without attached gender information.
The flu+antibiotics gave antibiotics orally between day1 and day3
```{r}
AD_healthy_male <- c(74,76,82,86,94,96,99,102,123,126,129,130,132,133,134,141)
AD_healthy_female <-c(72,87,88,89,91,97,100,114,116,119,120,124,125,127,128,138)
AD_male <- c(65,66,69,70,75,79,81,85,90,95,98,104,109,110,111,115,118,121,135,136,139,140)
AD_female <- c(67,68,71,73,77,78,80,83,84,92,93,101,103,105,106,107,108,112,113,117,122,131,137,142)
Hemo_healthy_female <- c(234,235)
Hemo_healthy_male <- c(231,232,233,236)
Hemo_disease_male <- c(213:218,220,221,223:226,230)
Hemo_disease_female <- c(219,222,227:229)
female_overweight <- c(185:198)
female_overweight_EGCG_treated <- c(199:205)
female_overweight_EGCG_Plus_treated <- c(206:212)
flu_healthy_control <- c(155,160,165,170,175,180)
flu_Plus_antibiotics_day1 <- c(156,161,166)
flu_Plus_antibiotics_day3 <- c(157,162,167)
flu_Plus_antibiotics_day7 <- c(158,163,168)
flu_Plus_antibiotics_day21 <- c(159,164,169)
flu_no_antibiotics_day1 <- c(171,176,181)
flu_no_antibiotics_day3 <- c(172,177,182)
flu_no_antibiotics_day7 <- c(173,178,183)
flu_no_antibiotics_day21 <- c(174,179,184)
MI_healthy_males <- c(143:148)
MI_males <- c(149:154)
tetanis_immunized <- c(59:64)

SamplesOnly <- all6[,c(1,AD_healthy_male,AD_healthy_female,AD_male,AD_female,
                       Hemo_healthy_female,Hemo_healthy_male,Hemo_disease_male,
                       Hemo_disease_female,
                       female_overweight,female_overweight_EGCG_treated,
                       female_overweight_EGCG_Plus_treated,
                       flu_healthy_control,flu_Plus_antibiotics_day1,
                       flu_Plus_antibiotics_day3,flu_Plus_antibiotics_day7,
                       flu_Plus_antibiotics_day21,flu_no_antibiotics_day1,
                       flu_no_antibiotics_day3,flu_no_antibiotics_day7,
                       flu_no_antibiotics_day21,MI_healthy_males,
                       MI_males,tetanis_immunized),]
row.names(SamplesOnly) <- SamplesOnly$GENE_SYMBOL
SamplesOnly2 <- SamplesOnly[,-1]
```

Now for making the class identifier as a vector to use in classifying these samples by type.
```{r}
AD_hm <- rep('AD Healthy Male',length(AD_healthy_male))
AD_hf <- rep('AD Healthy Female', length(AD_healthy_female))
AD_m <- rep('AD Male', length(AD_male))
AD_f <- rep('AD Female', length(AD_female))
Hemo_hf <- rep('Hemo Healthy Female', length(Hemo_healthy_female))
Hemo_hm <- rep('Hemo Healthy Male', length(Hemo_healthy_male))
Hemo_m <- rep('Hemo Male', length(Hemo_disease_male))
Hemo_f <- rep('Hemo Female', length(Hemo_disease_female))
overweight_female <- rep('Overweight Female', length(female_overweight))
overweight_female_EGCG <- rep('Overweight Female EGCG Treated',
                              length(female_overweight_EGCG_treated))
overweight_female_EGCG_Plus <- rep('Overweight Female EGCG+fishOil+VitC', 
                                   length(female_overweight_EGCG_Plus_treated))
flu_not_immunized_yet <- rep('Flu non immunized healthy',
                             length(flu_healthy_control))
flu_Plus_day1 <- rep('Flu Immunized Plus Antibiotics Day 1',
                     length(flu_Plus_antibiotics_day1))
flu_Plus_day3 <- rep('Flu Immunized Plus Antibiotics Day 3',
                     length(flu_Plus_antibiotics_day3))
flu_Plus_day7 <- rep('Flu Immunized Plus Antibiotics Day 7',
                     length(flu_Plus_antibiotics_day7))
flu_Plus_day21 <- rep('Flu Immunized Plus Antibiotics Day 21',
                      length(flu_Plus_antibiotics_day21))
flu_only_day1 <- rep('Flu Immunized Only Day 1', length(flu_no_antibiotics_day1))
flu_only_day3 <- rep('Flu Immunized Only Day 3', length(flu_no_antibiotics_day3))
flu_only_day7 <- rep('Flu Immunized Only Day 7', length(flu_no_antibiotics_day7))
flu_only_day21 <- rep('Flu Immunized Only Day 21', length(flu_no_antibiotics_day21))
MI_hm <- rep('MI Healthy Males', length(MI_healthy_males))
MI_m <- rep('MI Males', length(MI_males))
tet <- rep('Tetanis Immunized', length(tetanis_immunized))

ClassType <- as.data.frame(c(AD_hm, AD_hf, AD_m, AD_f, Hemo_hf, Hemo_hm, Hemo_m, Hemo_f,
               overweight_female, overweight_female_EGCG, overweight_female_EGCG_Plus,
               flu_not_immunized_yet, flu_Plus_day1, flu_Plus_day3, flu_Plus_day7,
               flu_Plus_day21, flu_only_day1, flu_only_day3, flu_only_day7,
               flu_only_day21, MI_hm, MI_m, tet))
colnames(ClassType) <- 'Class_Type'
Class2 <- as.data.frame(t(ClassType))
colnames(Class2) <- colnames(SamplesOnly2)

t(Class2) #Compare the row names when transposed to the assigned Class_Type

```

Replace the Samples GSM IDs with the class type labels.
```{r}
SamplesOnly3 <- SamplesOnly2
colnames(SamplesOnly3) <- ClassType$Class_Type
write.csv(SamplesOnly3, 'Dracula.csv', row.names=TRUE)

dim(SamplesOnly3)
colnames(SamplesOnly3)
classNames <- colnames(SamplesOnly3)
```

The above data was written to file and almost machine learning ready, but it would be a wide matrix of 12,845 features of gene symbols and 178 rows of class if transposed as is. So it will be filtered to the top FC genes from all6 for Alzheimer females and bottom genes of the same. No particular reason the genes with the most change in Alzheimer's Disease for females was selected, but it is known the brain controls many functions and AD patients die slowly by having body functions shut off until sadly and ultimately they cannot swallow their own saliva or breathe (many clients have had loved ones die of AD whom were told by nurses and doctors that this is how they die--maybe a 3rd party fact).
```{r}
AD_top125 <- all6[order(all6$Alz_FC_fem_ctrl_AD, decreasing=TRUE)[1:125],c(1,7)]
AD_bottom125 <- all6[order(all6$Alz_FC_fem_ctrl_AD)[1:125],c(1,7)]
AD250 <- rbind(AD_top125,AD_bottom125)
SamplesOnly3$Gene <- row.names(SamplesOnly3)
SamplesOnly4 <- merge(AD250,SamplesOnly3, by.x='GENE_SYMBOL', by.y='Gene')
row.names(SamplesOnly4) <- SamplesOnly4$GENE_SYMBOL
Samples250 <- SamplesOnly4[,-c(1:2)]

ML_250 <- as.data.frame(t(Samples250))
ML_250$Class_Type <- classNames
row.names(ML_250) <- NULL
ML_ready_250 <- ML_250[,c(251,1:250)]
```


